### ISSUE TITLE
when 'extension' clause context does not work for unpublished extensions

### ISSUE THREAD

**Issue by alberto-bc** [microsoft/vscode#151983]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.68.0
- OS Version: macOS Monterey 12.4

Reference: https://code.visualstudio.com/api/references/when-clause-contexts#available-contexts

extension | True when the extension's ID matches. Example: "extension == eamodio.gitlens".
-- | --

This does not work for extensions that are not published in the Marketplace. 

For our use case, we build and distribute an extension for private use. We would like to use `extension` in order to provide an easy way for users to open the full documentation:
```
 "extension/context": [
                {
                    "command": "extension.techCompany.openDocumentation",
                    "when": "extension == techCompany.privateExtensionId"
                }
            ],
```

The only workaround we found was to use `extensionStatus == installed`, but this will add a context entry for all extensions listed in the Marketplace list. This is not just misleading, but a limitation if we wanted to provide multiple private extensions

**Comment by sandy081** [comment]
It is working for me. I have the following context

```json
"extension/context": [
				{
					"command": "test.helloWorld",
					"when": "extension == sandy081.test-extension"
				}
			]
```

and I see it on my extension

<img width="589" alt="image" src="https://user-images.githubusercontent.com/10746682/173900316-ebe5ca80-17d2-48c9-886a-f80b6248f201.png">

This extension is not published.

**Comment by alberto-bc** [comment]
I performed more testing and I narrowed it down to the publisher's name: 
When using publisher `abc` the extension works as expected. When using uppercase `ABC` the entry is not showing up. 
Is there something I'm missing regarding publisher's name guidelines?
![abc](https://user-images.githubusercontent.com/66279626/174900294-922c0d92-faa7-4823-a921-515fd99017b9.jpg)
![upper_abc](https://user-images.githubusercontent.com/66279626/174900298-2fcacf9c-fd32-4fbd-879f-ef671f61f9e8.jpg)
